src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -4201,5 +4201,225 @@ static unsafe void M()
                 Diagnostic(ErrorCode.ERR_NoExplicitConv, "(delegate* managed<object?, long>)false").WithArguments("bool", "delegate*<object?, long>").WithLocation(9, 58)
                 );
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/81113")]
+        public void TestGetSymbolInfo_Direct()
+        {
+            var source = """
+                unsafe class C
+                {
+                    static void M2()
+                    {
+                        var d = (delegate*<void>)&C.M2;
+                        d();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilationWithFunctionPointers(source);
+            comp.VerifyEmitDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees.Single();
+            var semanticModel = comp.GetSemanticModel(syntaxTree);
+            var root = syntaxTree.GetRoot();
+
+            var addressOf = root.DescendantNodes().OfType<PrefixUnaryExpressionSyntax>().Single();
+            var symbolInfo = semanticModel.GetSymbolInfo(addressOf);
+            AssertEx.Equal("void C.M2()", symbolInfo.Symbol.ToTestDisplayString());
+            Assert.Empty(symbolInfo.CandidateSymbols);
+
+            var memberAccess = (MemberAccessExpressionSyntax)addressOf.Operand;
+            symbolInfo = semanticModel.GetSymbolInfo(memberAccess);
+            AssertEx.Equal("void C.M2()", symbolInfo.Symbol.ToTestDisplayString());
+            Assert.Empty(symbolInfo.CandidateSymbols);
+
+            symbolInfo = semanticModel.GetSymbolInfo(memberAccess.Name);
+            AssertEx.Equal("void C.M2()", symbolInfo.Symbol.ToTestDisplayString());
+            Assert.Empty(symbolInfo.CandidateSymbols);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/81113")]
+        public void TestGetSymbolInfo_Parethesized()
```

Comment:
There's a typo in test name. Also the test doesn't seem useful as parens generally do nothing. I'd remove this test

Reply:
The test is intentional to make sure we are walking through parentheses.

Reply:
I think the point is exactly that things work properly even in the presence of parens. We don't want a set of parens making it so that semantic info breaks again.